### DISCUSSION TITLE
Regular CI check of external plugins

### DISCUSSION THREAD

**Discussion by rsmmr** [zeek/zeek#1478] [Ideas]
It would be very helpful to automatically check external plugins against recent Zeek code changes, flagging things that break to both their authors and the Zeek development team. This idea keeps coming up in various contexts, it'd be great if we could figure out how to get this operational.

**Comment by JustinAzoff** [comment]
@grigorescu had something working here I think https://github.com/grigorescu/pkgci/blob/topic/vlad/github-actions/.github/workflows/ci_all.yml

I think the hard part ends up being that plugins often have external dependencies and it's not easy to build every plugin in the same environment without a well defined way of installing their dependencies beforehand.

**Comment by ckreibich** [comment]
Yes, this is nice work Vlad! I took this and made it use the Github action from zeek/zeek#1484, which means the whole thing gets a bit easier. You can see the resulting YAML [here](https://github.com/ckreibich/packages/blob/topic/christian/ci/.github/workflows/nightly.yaml) and a sample run [here](https://github.com/ckreibich/packages/actions/runs/707379394).

I'm currently skipping packages with `external_depends`. One thing that isn't there yet is a notification mechanism for broken packages. Some sort of dashboard rendering of those workflows would also be nice — I looked around a bit but couldn't find anything terribly promising.

Thoughts?

**Comment by ckreibich** [comment]
A comment on the approach: the matrix strategy for parallelizing jobs [tops out at 256 entries](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix). We're currently at ~160 packages depending on how/whether we filter, so there's some runway left. We can then set up multiple jobs that separate packages into sets no larger than 256 each. So I think this'll be fine for the foreseeable future.
